app/uk/gov/hmrc/fileupload/controllers/EnvelopeController.scala

```diff
@@ -27,27 +29,41 @@ import uk.gov.hmrc.fileupload.utils.JsonUtils.jsonBodyParser
 import uk.gov.hmrc.fileupload.write.envelope._
 import uk.gov.hmrc.fileupload.write.infrastructure._
 import uk.gov.hmrc.fileupload.{EnvelopeId, FileId, _}
-import uk.gov.hmrc.play.microservice.controller.BaseController
 
 import scala.concurrent.{ExecutionContext, Future}
 import scala.language.postfixOps
 
+
+package object envelope {
+
+  import play.api.libs.concurrent.Execution.Implicits._
+
+  object EnvelopeController extends EnvelopeController(
+    withBasicAuth = MicroserviceGlobal.withBasicAuth,
```

Comment:
We had to do away with getControllerInstance, that meant either creating objects or using Guice to create the controllers with DI. Guice is not a bad idea but if we do that then I would like to wire everything up with it to be consistent, perhaps a bit of a big step for now. The baby step we chose to take was to replace the lazy vals in MicroserviceGlobal with objects and use Play's StaticRoutesGenerator for controllers. We initially tried to define the objects in the same file as MicroserviceGlobal as has been done in the FE see....

https://github.com/opetch/file-upload-frontend/blob/f97a4c82ed9886eec82784c803c8b3a6fd11683f/app/uk/gov/hmrc/fileupload/frontendGlobal.scala

.... the reason we moved away from that in the BE was because we we're having issues with the ReverseControllers that play generates, we couldn't have them in the same package as one another. So we moved the into their respective classes files as companion objects this has since changed slightly as we have the package objects to protect the class from the execution context import. We could easily move the package objects that they are wrapped in into the global file, that will fix the up hierarchy reference, that can be done in the morning if you think it would be nicer?